doc/users-guide/users-guide-crypto.adoc

```diff
@@ -61,6 +90,50 @@ session’s completion queue, which can be accessed directly or via the ODP
 scheduler. The completion event contains the status of the operation and the
 result. The application has the responsibility to free the completion event.
 
+==== Packet-based Crypto Operations
+To simplify the original cryptographic operation request API, as well as to
+be more flexible and consistent with the protocol-aware APIs introduced for
+IPsec support, a newer packet-oriented set of cryptographic operation
+APIs is also provided. Applications may use either API set, but going forward
+it is expected that these newer APIs will be the focus of continued
+development.
+
+Instead of a single `odp_crypto_operation()` API, the packet-based APIs
+provide two APIs. `odp_crypto_op()` is the synchronous form while
+`odp_crypto_op_enq()` is the asynchronous form.
+
+Both forms take an input array of packets, an optional output array of packets
+to receive the results, and an `odp_crypto_packet_op_param_t` struct that
+describes the operation to be performed. As with the original APIs, the
+output array may be the same packets to request in-place operation, or may
+be specified as `ODP_PACKET_INVALID` to request that ODP allocate output
+packets from the pool associated with the `odp_crypto_session_t` being used.
+
+The key differences between the `odp_crypto_op_param_t` used by the original
+APIs and the `odp_crypto_packet_op_param_t` used by the new APIs are:
+
+* The original API takes a single `odp_crypto_op_param_t` since it operates on
+a single packet whereas the new forms take an array of
+`odp_crypto_packet_op_param_t` structs, one for each input packet.
+
+* The `odp_crypto_packet_op_t` does not contain any packet information since
+the input and output packets are supplied as API parameters rather than being
+encoded in this struct.
+
+* The `odp_crypto_packet_op_t` does not contain a user context field.
+
+In addition, the `odp_crypto_session_t` field `op_mode` is used instead of
+the `pref_mode` field when the packet-oriented APIs are used. If the
+`op_mode` is set to `ODP_CRYPTO_SYNC` then the synchronous form of the API
+must be used and if `op_mode` is set to `ODP_CRYPTO_ASYNC` then the
+asynchronous form of the API must be used.
+
+As with the original APIs, an asynchrous completion is indicated by an
+`ODP_EVENT_CRYPTO_COMPL` being sent to the `output_queue` associated with
```

Comment:
No-no-no, this is incorrect. Assynchronous completion is indicated by an event 
 with type/subtype being set to `ODP_EVENT_PACKET`/`ODP_EVENT_PACKET_CRYPTO`. New API does not use `ODP_EVENT_CRYPTO_COMPL` at all!